src/middlewared/middlewared/plugins/cluster_linux/ctdb_private.py

```
@@ -0,0 +1,241 @@
+                f.write(pri_ip + '\n')
+                f.flush()
+        else:
+            with open(PRI_IP_FILE) as f:
+                lines = f.read().splitlines()
+
+            # in the case of removing a node, a "#"
+            # _MUST_ be put in front of the entry in
+            # the file and the other lines _MUST_ be
+            # kept in the same order when writing it
+            # back out or ctdb will barf and fail to
+            # reload...
+            index = lines.index(pri_ip)
+            lines[index] = '#' + pri_ip
+
+            with open(PRI_IP_FILE, 'w') as f:
+                f.writelines(map(lambda x: x + '\n', lines))
+                f.flush()
+
+        return True
+
+    @accepts(Dict(
+        'node_create',
+        IPAddr('ip'),
+    ))
+    @job(lock=PRI_LOCK)
+    async def create(self, job, data):
+        """
+        Add a ctdb private address to the cluster
+
+        `ip` is an IP v4/v6 address
+        """
+
+        schema_name = 'node_create'
+        verrors = ValidationErrors()
+
+        await self.middleware.call(
+            'ctdb.private.ips.common_validation', data['ip'], schema_name, verrors
+        )
+        if not await self.middleware.call('ctdb.private.ips.update_file', data['ip'], verrors):
+            raise CallError(f'Failed updating {PRI_IP_FILE}. Please check logs.')
+
+        return data['ip']
+
+    @accepts(Dict(
+        'node_delete',
+        IPAddr('ip'),
+    ))
+    @job(lock=PRI_LOCK)
+    async def delete(self, job, data):
+        """
+        Delete a Private IP address from the ctdb cluster.
+
+        `ip` is an IP v4/v6 address
+        """
+
+        schema_name = 'node_delete'
+        verrors = ValidationErrors()
+
+        await self.middleware.call(
+            'ctdb.private.ips.common_validation', data['ip'], schema_name, verrors, True
+        )
+        if not await self.middleware.call('ctdb.private.ips.update_file', data['ip'], verrors, True):
```

Wouldn't it be better to make ctdb.private.ips.update_file raise an error show we could potentially know the error instead of telling user to check logs?